tests/subsys/net/lib/nrf_cloud/cloud/src/main.c

```diff
@@ -1478,6 +1478,28 @@ ZTEST(nrf_cloud_test, test_cloud_send_invalid_topic_type)
 	zassert_equal(-ENODATA, ret, "return should be -ENODATA with invalid topic type");
 }
 
+/* Verify that nrf_cloud_send will fail if object encoding fails */
+ZTEST(nrf_cloud_test, test_cloud_send_obj_encode_fail)
```

Comment:
I think there needs to be a commit between this commit and the next one that includes unit tests for each of the newly added functions -- That way we don't wind up with an increasing backlog of non-verified functions

Just some basic pass/fail tests of each of the modifier functions would suffice, I think

I'm not sure if a test suite for nrf_cloud_codec.c already exists, but if not, it would be, IMO, worthwhile to create one now and just populate it with tests for the new functions for the time being

Reply:
yes, tests will need to be written for `nrf_cloud_codec.c`, a separate ticket exists: IRIS-4788

Reply:
My interpretation of htat ticket is that it's for retroactively creating unit tests for the functions that already exist in `nrf_cloud_codec`, as opposed to the functions added in this PR. I think we want to move toward implementing unit tests for functions as we add them, right?

Reply:
yes that is the goal. 
it will probably take me an entire sprint to implement tests for the cloud codec since i haven't written tests in a very long time.
i'll defer to @plskeggs